### PR TITLE
[PHP-80]3.0.0 landingi/core proposal

### PR DESCRIPTION
**Pull request by Ferror**
# READ BEFORE REVIEW

A totally new approach to landingi/core.

## BREAKING CHANGES
* CI/CD in Travis CI
* Lack of Domain/Application/Infrastructure separation
* No Context classes
* Core as storage for reusable classes between services.
* Remove old classes
* PHP 7.4
* docker-compose for development
* Modern CI - phpstan, phpunit, ecs

I encourage to review in github or in IDE
Screenshot 2021-01-28 at 16 47 37

### CODE REVIEW

**Comment by nexik** [comment]
1. PR is too big for my standard
2. first I would introduce CI I think Travis for Open Source repo is the best solution.
3. If we are planning to go to the new version for landingi/core I would recommend going to PHP 8
4. Why removing LICENCE. Do you plan to go to a private library? I think we should stay open.
5. Empty .gitignore file in the folder with other files is meaningless. This pattern is used when you want to add empty folders to git because git doesn't accept them (empty folders).
6. Adding supervisord to library repo is required?  Especially after removing Infrastructure and Domain namespace this repo will be a simple SPL library for the Landingi organization

**Comment by Ferror** [comment]
> 3. If we are planning to go to the new version for landingi/core I would recommend going to PHP 8

Providing a new bundle based on PHP 8 is another blockage for our system as 4 (I don't count application) of our repositories/services don't have even PHP 7.4 and they all use AWS. The transition to PHP 8 will take time. The goal of this core should be to automate and structurize the AWS Clients and most common VO/Systems like FF, Event Store, Pagination, Responses etc.

> 2. first I would introduce CI I think Travis for Open Source repo is the best solution.
> 4. Why removing LICENCE. Do you plan to go to a private library? I think we should stay open.

The current `landingi/core` has no **open source value** as we provide the only **internal** classes. To provide to our customers aka Integrators SDK/Bundle we should prepare PHP/Ruby/JS clients for our public API. Therefore LICENSE and TravisCI are not required.


============================
Other stuff will be fixed

**Comment by Ferror** [comment]
Pozbyłem się gitignore'a i supervisora. Usunąłem też EventStore, żeby zmniejszyć PR - w tym momencie jest kilka klas dla AWS, czyli to co faktycznie się przyda developerom - wszędzie tego używamy i wszędzie jest inny config :)

**Comment by nexik** [comment]
PR is still too big (10k changes). English, please :)

**Comment by Ferror** [comment]
@nexik I have provided the requested changes, but still, the `composer.lock` file takes 7k changes due to PHP 7.4 update.

But the repo is smooth and waiting for other improvements/classes.

Screenshot 2021-02-03 at 15 12 02

**Comment by devofdisaster** [comment]
My only question is whether this is supposed to be a general open source library, or a Landingi-specific library that happens to be open source. 
The proposed `Currency` and `Language` classes are limited to a subset of currencies and languages used within the context of the Landingi platform, making them as useful as everything in the "old" landingi/core.
This makes sense in the second case of course (if we are to use it both internally and for example as an OS base for a PHP SDK), but I think this should be made clear in the readme file.

File: src/Currency.php [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+    private const CURRENCIES_WITH_LEFT_SIDE_SYMBOL = [self::GBP, self::USD, self::BRL];
+
+    private string $code;
+
+    /**
+     * @throws \InvalidArgumentException
+     */
+    public function __construct(string $code)
+    {
+        if (! \in_array($code, [self::USD, self::PLN, self::RUB, self::EUR, self::GBP, self::BRL], true)) {
```
Comment (SBielawski) [inline src/Currency.php:47]: `use function in_array` :)

Comment (Ferror) [reply src/Currency.php:47]: ;p

**Review by devofdisaster — APPROVED** [review]
Ok, makes sense to me

**Pull request merged**